### ISSUE TITLE
Show user ID in the profile, small and grey somewhere underneath

### ISSUE THREAD

**Issue by Tirokk** [Human-Connection/Nitro-Web#193]
**Is your feature request related to a problem? Please describe.**
At the moment one has problems to identify a user safely.
If you have fake users or trouble with one, it is difficult to see if this is still the same profile.

**Describe the solution you'd like**
Show user ID in the profile somewhere underneath, small and grey. Will say not really prominent. If you get trouble you can make a screenshot of the profile.

**Describe alternatives you've considered**
You might kind of see the change of a users name if you remember a comment or so somewhere which  seems to change it’s name. But this is really unsafe.

**Additional context**
A user ID showing up in profiles we use on PlanetRomeo and this is kind of effective.

**Comment by appinteractive** [comment]
Would the slug not be enough?

**Comment by roschaefer** [comment]
@appinteractive I talked to @Tirokk on our regional user meetup in Berlin on Saturday. The user ID, slug and name have the following purposes and properties:

| | Modifiable | Referenceable | Unique | Purpose |
| -- | -- | -- | --| -- |
| ID | no | yes | yes | Identity, Traceability, Links |
| Slug | yes | yes | yes | @ - Mentions, SEO-friendly URL |
| Name | yes | no | no | Search, self-description |

Ie. the slugs behave in the same way as on Github and Twitter: You can change it but there must not be any other account with the same slug. You would also mention (and remember!) another user account via the slug and not the username. 

We could also include the id in links to a user account. That way, if the user decides to change her slug, the old links will not point to a totally different user account at some point.

`Referenceable` in the table above would mean, you could type in the url `/users/roschaefer` and the server would redirect to `/users/4711-roschaefer`.

**Comment by Tirokk** [comment]
… or you get a list of all users with the name roschaefer, if there are more than one @roschaefer @appinteractive .

**Comment by roschaefer** [comment]
@tirokk no, @roschaefer is my slug, not my name. So there must be only one user account with that slug at a given time.

@tirokk could you research what happens to the old links if I would change my slug on Github?

**Comment by Tirokk** [comment]
Than you have to make the readable part in the slug sole too, by adding a number to it @roschaefer ?
Why, if there is an ID behind it?
To distinguish between people and for readable clarification?

So if I would like the name **roschaefer** as well, it would become **roschaefer1** automatically?
Or I get a message that **roschaefer** already exists and have to choose an other one?

I research …

**Comment by Tirokk** [comment]
Because GitHub don’t use IDs in the Slug …

The impact if one changes the username is diverse on GitHub:
From redirection of repository links as long as no one else uses your old usernames to commit references and **@mentions** to your user profile breaking … etc.

> When you change your GitHub username, most references to your repositories under the old username automatically change to the new username. However, some links to your profile won't automatically redirect. Commits made with your username@users.noreply.github.com email address will no longer be associated with your account. Commits made using your ID+username@users.noreply.github.com email address will continue to be associated with your account.

More details see here [What happens when I change my username?](https://help.github.com/en/articles/what-happens-when-i-change-my-username)

**Comment by appinteractive** [comment]
That would also be my suggestion. Also could we limit the the amount of possible changes in a given timeframe (once a week/month or so) to reduce abuse.

**Comment by Tirokk** [comment]
That’s not me. @appinteractive 
Computer are made to help humans and therefore made to working correctly.
It’s not only about abuse. It’s about good functionality.

Hope I got you right … 😉

Broken links are not acceptable for me. (I can accept, if you vote me out, but would not be my style 😂)

**Comment by Tirokk** [comment]
To limit the amount of changes for a username is something to think about. @appinteractive 

To limit the the change amounts of a post header seems not to be possible. It can violate rules and has to be changed immediately. Maybe more then once a week or so.
